File: plugins/on_audio_query-2.6.1/android/src/main/kotlin/com/lucasjosino/on_audio_query/controller/OnAudioController.kt
package com.lucasjosino.on_audio_query.controller

import android.content.Context
import android.util.Log
import com.lucasjosino.on_audio_query.query.*
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

class OnAudioController(
    private val context: Context,
    private val call: MethodCall,
    private val result: MethodChannel.Result
) {

    //
    fun onAudioController() {
        when (call.method) {
            //Query methods
            "querySongs" -> OnAudiosQuery().querySongs(context, result, call)
            "queryAlbums" -> OnAlbumsQuery().queryAlbums(context, result, call)
            "queryArtists" -> OnArtistsQuery().queryArtists(context, result, call)
            "queryPlaylists" -> OnPlaylistQuery().queryPlaylists(context, result, call)
            "queryGenres" -> OnGenresQuery().queryGenres(context, result, call)
            "queryArtwork" -> OnArtworksQuery().queryArtwork(context, result, call)
            "queryAudiosFrom" -> OnAudiosFromQuery().querySongsFrom(context, result, call)
            "queryWithFilters" -> OnWithFiltersQuery().queryWithFilters(context, result, call)
            "queryAllPath" -> OnAllPathQuery().queryAllPath(context, result)
            //Playlists methods
            "createPlaylist" -> OnPlaylistsController().createPlaylist(context, result, call)
            "removePlaylist" -> OnPlaylistsController().removePlaylist(context, result, call)
            "addToPlaylist" -> OnPlaylistsController().addToPlaylist(context, result, call)
            "removeFromPlaylist" -> OnPlaylistsController().removeFromPlaylist(
                context,
                result,
                call
            )
            "renamePlaylist" -> OnPlaylistsController().renamePlaylist(context, result, call)
            "moveItemTo" -> OnPlaylistsController().moveItemTo(context, result, call)
        }
    }
}

File: plugins/on_audio_query-2.6.1/ios/Classes/controller/OnAudioController.swift
import Flutter

public class OnAudioController {
    var call: FlutterMethodCall
    var result: FlutterResult
    
    init(call: FlutterMethodCall, result: @escaping FlutterResult) {
        self.call = call
        self.result = result
    }
    
    
    //This method will sort call according to request.
    public func chooseMethod() {
        // All necessary method to this plugin support both platforms, only playlists
        // are limited when using [IOS].
        switch call.method {
        case "querySongs":
            OnAudioQuery(call: call, result: result).querySongs()
        case "queryAlbums":
            OnAlbumsQuery(call: call, result: result).queryAlbums()
        case "queryArtists":
            OnArtistsQuery(call: call, result: result).queryArtists()
        case "queryGenres":
            OnGenresQuery(call: call, result: result).queryGenres()
        case "queryPlaylists":
            OnPlaylistsQuery(call: call, result: result).queryPlaylists()
        case "queryAudiosFrom":
            OnAudiosFromQuery(call: call, result: result).queryAudiosFrom()
        case "queryWithFilters":
            OnWithFiltersQuery(call: call, result: result).queryWithFilters()
        case "queryArtwork":
            OnArtworkQuery(call: call, result: result).queryArtwork()
        // The playlist for [IOS] is completely limited, the developer can only:
        //   * Create playlist
        //   * Add item to playlist (Unsuported, for now)
        //
        // Missing methods:
        //   * Rename playlist
        //   * Remove playlist
        //   * Remove item from playlist
        //   * Move item inside playlist
        case "createPlaylist":
            OnPlaylistsController(call: call, result: result).createPlaylist()
        case "addToPlaylist":
            OnPlaylistsController(call: call, result: result).addToPlaylist()
        default:
            // All non suported methods will throw this error.
            result(FlutterMethodNotImplemented)
        }
    }
}
